airbyte-webapp/src/views/Connector/ConnectorCard/components/Controls.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Button } from "components/ui/Button";
+import { FlexContainer, FlexItem } from "components/ui/Flex";
+
+import { SynchronousJobRead } from "core/request/AirbyteClient";
+
+import { TestCard } from "./TestCard";
+
+interface IProps {
+  formType: "source" | "destination";
+  isSubmitting: boolean;
+  isValid: boolean;
+  dirty: boolean;
+  onCancelClick: () => void;
+  onDeleteClick?: () => void;
+  onRetestClick: () => void;
+  onCancelTesting: () => void;
+  isTestConnectionInProgress?: boolean;
+  errorMessage?: React.ReactNode;
+  job?: SynchronousJobRead;
+  connectionTestSuccess: boolean;
+  hasDefinition: boolean;
+  isEditMode: boolean;
+}
+
+export const Controls: React.FC<IProps> = ({
+  isTestConnectionInProgress,
+  isSubmitting,
+  formType,
+  hasDefinition,
+  isEditMode,
+  dirty,
+  onDeleteClick,
+  onCancelClick,
+  ...restProps
+}) => {
+  return (
+    <>
+      {hasDefinition && (
+        <TestCard
+          {...restProps}
+          formType={formType}
+          isTestConnectionInProgress={isTestConnectionInProgress}
+          isEditMode={isEditMode}
+          dirty={dirty}
+        />
+      )}
+      <FlexContainer>
+        <FlexItem grow>
+          {isEditMode && (
+            <Button variant="danger" type="button" onClick={onDeleteClick} data-id="open-delete-modal">
+              <FormattedMessage id={`tables.${formType}Delete`} />
+            </Button>
+          )}
+        </FlexItem>
+        {isEditMode && (
+          <Button type="button" variant="secondary" disabled={isSubmitting || !dirty} onClick={onCancelClick}>
+            <FormattedMessage id="form.cancel" />
+          </Button>
+        )}
+        <Button type="submit" disabled={isSubmitting || !dirty}>
```

Comment:
You'll need to fix this `disabled` logic in a similar way, so that if we arent in edit mode, the form can still be submitted even if its not dirty (see above comment)

Reply:
Adjustted